src/test/kotlin/step2/StringCalculatorTest.kt

```diff
@@ -3,26 +3,30 @@ package step2
 import io.kotest.assertions.throwables.shouldNotThrowAny
 import io.kotest.assertions.throwables.shouldThrow
 import io.kotest.core.spec.style.ShouldSpec
+import io.kotest.datatest.withData
+import io.kotest.inspectors.forAll
 import io.kotest.matchers.shouldBe
 
 class StringCalculatorTest : ShouldSpec({
     context("InputValidation") {
         should("throw exception if null or empty") {
-            shouldThrow<IllegalArgumentException> {
-                StringCalculator("")
+            listOf("", " ", null).forAll {
+                shouldThrow<IllegalArgumentException> {
+                    StringCalculator("")
+                }
             }
         }
 
         should("throw exception if split strings are less than 3") {
-            listOf("12 34", "12 +", "+ +").forEach {
+            listOf("12 34", "12 +", "+ +").forAll {
                 shouldThrow<IllegalArgumentException> {
                     StringCalculator(it)
                 }
             }
         }
 
         should("contain at least 3 split strings") {
-            listOf("12 + 56", "12  + 56", "12 + 34", "12 + 34").forEach {
+            listOf("12 + 56", "12  + 56", "12 + 34", "12 + 34").forAll {
```

Comment:
> How about adding a value to listOf that causes this test to fail? (e.g., 12 + 1)
When the test failed, were you able to quickly identify which input caused the failure?
I recommend checking out Kotest's Data-Driven Testing or Inspectors, which can help with this!

Oh, actually, I did find it difficult while testing for this. It was hard to figure out why a test was failing while writing it myself... I was going to look up whether there was a function like assertAll later, but I completely forgot. Thanks for pointing it out! 😆 I think it'll make testing much easier!